samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -585,35 +577,32 @@ object SamzaContainer extends Logging {
 
             // We use the logged storage base directory for change logged and side input stores since side input stores
             // dont have changelog configured.
-            val storeDir = if (changeLogSystemStreamPartition != null || sideInputStoresToSystemStreams.contains(storeName)) {
-              TaskStorageManager.getStorePartitionDir(loggedStorageBaseDir, storeName, taskName)
+            val storeDir = if (sideInputStoresToSystemStreams.contains(storeName)) {
```

Comment:
Since we have a `filter()` earlier, it looks like this code-path is reached only for sideInputStores? If so, do we still need the `if` check here?

Reply:
done